app/controllers/charge_controller.js

```diff
@@ -129,81 +127,72 @@ module.exports = {
 
     if (charge.status === State.AUTH_READY) return redirect(res).toAuthWaiting(req.chargeId)
     // else
-    AWSXRay.captureAsyncFunc('chargeValidator_verify', function (subSegment) {
-      validator.verify(req)
-        .catch(() => {
-          subSegment.close('error')
-          redirect(res).toNew(req.chargeId)
-        })
-        .then(data => {
-          subSegment.close()
-          card = data.card
+    validator.verify(req)
+      .catch(() => {
+        redirect(res).toNew(req.chargeId)
+      })
+      .then(async data => {
```

Comment:
I'm not sure I like mixing the `.then(...)` and async/await syntax in the same function and maybe in the same class. But if we're incrementally improving, then maybe doing it for the entire enclosing function would be the best way to go.

Reply:
I agree. I tried to make `validator.verify(req)` async but I wasn't quite sure how. maybe you could help me out with that.